user-ops-indexer/user-ops-indexer-logic/src/indexer/base_indexer.rs

```diff
@@ -216,9 +216,16 @@ impl<L: IndexerLogic + Sync> Indexer<L> {
             .filter_map(|tx_hash| async move { tx_hash });
 
         stream_txs
-            .for_each_concurrent(Some(self.settings.concurrency as usize), |tx| async move {
+            .map(Ok)
+            .try_for_each_concurrent(Some(self.settings.concurrency as usize), |tx| async move {
                 let mut backoff = vec![5, 20, 120].into_iter().map(Duration::from_secs);
-                while let Err(err) = &self.handle_tx(tx, variant).await {
+                while let Err(err) = self.handle_tx(tx, variant).await {
+                    // terminate stream if WS connection is closed, indexer will be restarted
+                    if self.client.as_ref().supports_subscriptions() && err.to_string() == WsClientError::UnexpectedClose.to_string() {
```

Comment:
Nit: just interested if you've checked this works in practice? Doesn't `anyhow` add anything additional to the message that changes it?

In theory, it seems that you may also try to downcast `anyhow::Error` to the `WsClientError` to compare it in a more direct way. But I'm also not sure that would work, so probably better to leave as it is, if you'd checked current approach
```suggestion
                    if self.client.as_ref().supports_subscriptions()
                        && err.downcast_ref::<WsClientError>().is_some_and(|err| matches!(err, WsClientError::UnexpectedClose)) {
```

Reply:
Yes, I've run a manual test to confirm anyhow wrapper converts to string without anything extra.

I've tried to fight with downcast, but the problem is that it's not WsClientError wrapped in anyhow, it's actually first wrapped in another Enum with `Box<dyn ...>`, and then in anyhow `Box<dyn ...>`. Haven't figured out a good and clean way to double-unwrap it. So gave up on downcasting.

Reply:
The final result was `anyhow::Error(ProviderError::JsonRpcClientError(Box::new(WsClientError::UnexpectedClose)))`